lib/scraper.js

```diff
@@ -99,6 +99,10 @@ Scraper.prototype.getResourceHandler = function getHandler (resource) {
 Scraper.prototype.loadResource = function loadResource (resource) {
 	var self = this;
 
+	if(!self.options.urlFilter(resource.url)){
+		return Promise.resolve(null);
```

Comment:
The problem with the current code is that the following things are too tightly coupled:

1. Reading resource urls from the code (HTML, CSS).
2. Determining whether the urls should be processed (`urlFilter`).
3. Determining the location where the file should be saved.
4. Retrieving and saving the resources.
5. Updating the urls based on the location in which they where saved.

To properly solve this, the code would need huge refactoring. You would want to end up with something like this (using ES6 syntax for brevity):
``` javascript
resource
  .createChildResources() // reads the child resource urls using the right handler, based on the resource type, and constructs resources from them.
  .then(childResources => _.filter(childResources, childResource => context.options.urlFilter(childResource.url)));
  .then(childResources => _.forEach(childResource => childResource.updateUrlInParent()))
  .then(childResources, Promise.all(_.map(childResources, childResource => loadAndSaveResource(childResource)))
```